Core/Utilities/List.py

```diff
@@ -127,7 +127,8 @@ def breakListIntoChunks(aList, chunkSize):
   """
   if chunkSize < 1:
     raise RuntimeError("chunkSize cannot be less than 1")
-  if isinstance(aList, (set, dict, tuple, six.moves.collections_abc.Iterable)):
+  if isinstance(aList, (set, dict, tuple, {}.keys().__class__,
```

Comment:
why ? Also if there is a good reason, you may want to document it, because someone that knows of `collections_abc` and comes after you will surely modify it back :-D

Reply:
I added the `six.moves.collections_abc.Iterable` in an earlier commit but in integration it's currently:

https://github.com/DIRACGrid/DIRAC/blob/34c753dc0cd827d7df262cd1989bc7adfbb05f1c/Core/Utilities/List.py#L130-L131

It needs to be changed as `dict_keys`, `dict_values` and `dict_items` are all distinct types in Python 3.

My change to check against `Iterable` seemed nicer but it was a stupid thing to do. (`str` is iterable but `list("LFN:/myfile.txt")` is not what I wanted ;))

Reply:
> It needs to be changed as dict_keys, dict_values and dict_items are all distinct types in Python 3.
The more I learn about Python 3, the less of an improvement it looks to me :thinking: 

Anyway, I understand now. Ahhhhhh if only we had unit tests :-p

Maybe a bit less hacky and more understandable way ? 

```
if isinstance(aList, six.moves.collections_abc.Iterable) and not isinstance(aList, six.string_types):
```

Reply:
Well the code should be modernised everywhere and this function should be deleted (`[::chunkSize]`) but that should be a separate PR. I prefer it like this as it definitely won't change how DIRAC currently works in Python 2.

Reply:
yep